### ISSUE TITLE
Extraneous input '<EOF>'

### ISSUE THREAD

**Issue by marcospassos** [antlr/intellij-plugin-v4#324]
We have an use case where we need to match a ';' or the end of the file. The following simplified grammar works great, matching either `foo`,  `foo; or foo` or `foo; or foo;`:

```antlr4
grammar Example;

root: conjunction EOF ;
conjunction: conjunction 'or' conjunction | atom ;
atom: 'foo' ( ';' | EOF ) ;

WS: [ \n\t\r]+ -> channel(HIDDEN);
```

However, the plugin reports the following error which does not exist in fact:

![image](https://user-images.githubusercontent.com/943036/34632459-af20b81c-f25c-11e7-8edf-8eda4e0264fa.png)

In contrast to:

```
>java  org.antlr.v4.gui.TestRig Example root -tree
> foo
(root (conjunction (atom foo <EOF>)) <EOF>)
```

![image](https://user-images.githubusercontent.com/943036/34632878-54767fca-f25f-11e7-904d-7de667986ba8.png)

**Comment by bjansen** [comment]
@parrt I suspect it's a limitation/bug in `org.antlr.v4.runtime.ParserInterpreter`, wdyt?

**Comment by parrt** [comment]
@bjansen yeah, must be. Though, i did create that for use with the plugin i think ;)  hmm...i don't have the energy to track that down.
